nrf_security/src/mbedcrypto_glue/CMakeLists.txt

```diff
@@ -32,6 +32,8 @@ if(CONFIG_NRF_CRYPTO_GLUE_LIBRARY)
   #
   zephyr_library_named(mbedcrypto_glue)
 
+  zephyr_library_compile_options_ifdef(CONFIG_SHORT_WCHAR_T -fshort-wchar)
+
```

Comment:
This should be replaced with

zephyr_compile_options_ifdef(CONFIG_SHORT_WCHAR_T -fshort-wchar)

The same for other zephyr_library_'s.

The zephyr_compile_options invocation should be placed in upstream zephyr. If there is a need to get it in early it can in addition be placed in ncs/nrf/CMakeLists.txt until the upstream patch has made it's way downstream and made the nrf patch redundant.

The same applies to config SHORT_WCHAR_T.